services/notify/src/api.js

```diff
@@ -86,6 +114,110 @@ builder.declare({
+  scopes: 'notify:manage-blacklist:<notificationType>/<notificationAddress>',
+  input: 'notification-address.yml',
+  title: 'Blacklist Given Address',
+  description: [
+    'Add the given address to the notification blacklist. The address',
+    'can be of either of the three supported address type namely pulse, email',
+    'or IRC(user or channel). Addresses in the blacklist will be ignored',
+    'by the notification service.',
+  ].join('\n'),
+}, async function(req, res) {
+  // The address to blacklist
+  let address = Object.assign({}, {
+    notificationType: req.body.notificationType,
+    notificationAddress: req.body.notificationAddress,
+  });
+
+  await req.authorize(req.body);
+  try {
+    await this.BlacklistedNotification.create(address);
+  } catch (e) {
+    if (e.name !== 'EntityAlreadyExistsError') {
+      throw e;
+    }
+  }
+  res.sendStatus(200);
+});
+
+builder.declare({
+  method: 'delete',
+  route: '/blacklist/delete',
+  name: 'deleteBlacklistAddress',
+  scopes: 'notify:manage-blacklist:<notificationType>/<notificationAddress>',
+  input: 'notification-address.yml',
+  title: 'Delete Blacklisted Address',
+  description: [
+    'Delete the specified address from the notification blacklist.',
+  ].join('\n'),
+}, async function(req, res) {
+  // The address to remove from the blacklist
+  let address = {
+    notificationType: req.body.notificationType,
+    notificationAddress: req.body.notificationAddress,
+  };
+
+  await req.authorize(req.body);
+  try {
+    await this.BlacklistedNotification.remove(address);
+  } catch (e) {
+    if (e.name !== 'ResourceNotFoundError') {
+      throw e;
+    }
+  }
   res.sendStatus(200);
 });
+
+builder.declare({
+  method: 'get',
+  route: '/blacklist/list',
+  name: 'list',
+  output: 'notification-address-list.yml',
+  title: 'List Blacklisted Notifications',
+  query: {
+    continuationToken: Entity.continuationTokenPattern,
+    limit: /^[0-9]+$/,
+  },
+  description: [
+    'Lists all the blacklisted addresses.',
+    '',
+    'By default this end-point will try to return up to 1000 addresses in one',
+    'request. But it **may return less**, even if more tasks are available.',
+    'It may also return a `continuationToken` even though there are no more',
+    'results. However, you can only be sure to have seen all results if you',
+    'keep calling `list` with the last `continuationToken` until you',
+    'get a result without a `continuationToken`.',
+    '',
+    'If you are not interested in listing all the members at once, you may',
+    'use the query-string option `limit` to return fewer.',
+  ].join('\n'),
+}, async function(req, res) {
+  const continuation = req.query.continuationToken || null;
+  const limit = Math.min(parseInt(req.query.limit || 1000, 10), 1000);
+  const query = await this.BlacklistedNotification.scan({}, {continuation, limit});
+
+  return res.reply({
+    addresses: query.entries.map(address => address._properties),
```

Comment:
It's probably better to construct this explicitly, rather than relying on `_properties` (which is internal, after all).  So `address => {notificationType: address.notificationType, ..}`